extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryRecorder.java

```diff
@@ -93,11 +93,17 @@ private Map<String, String> getOtelConfigs() {
                 // load new properties
                 for (String propertyName : config.getPropertyNames()) {
                     if (propertyName.startsWith("quarkus.otel.")) {
-                        String value = config.getValue(propertyName, String.class);
-                        if (propertyName.endsWith("timeout") || propertyName.endsWith("delay")) {
-                            value = OTelDurationConverter.INSTANCE.convert(value);
+                        ConfigValue value = config.getConfigValue(propertyName);
+                        if (value.getValue() != null) {
+                            if (propertyName.endsWith("timeout") || propertyName.endsWith("delay")) {
+                                quarkus.put(propertyName.substring(8),
+                                        OTelDurationConverter.INSTANCE.convert(value.getValue()));
+                            } else {
+                                quarkus.put(propertyName.substring(8), value.getValue());
+                            }
+                        } else if (value.getValue() == null && value.getRawValue() != null) {
+                            config.getValue(propertyName, String.class);
```

Comment:
Wonder whether this should be
```java
quarkus.put(propertyName.substring(8), config.getValue(propertyName, String.class));
```

Reply:
This is OK.

We now provide relocations and fallbacks from specific path trees `quarkus.otel.exporter.otlp.traces` and `quarkus.otel.exporter.otlp.metrics` to `quarkus.otel.exporter.otlp`. The relocations and fallbacks have to include the mapping names to work correctly. Because we don't know if we will get the specific or the base configuration, the properties are marked as `Optional` in the mapping interface.

In this code, we copy the `quarkus` properties to the `otel` properties. Because we now have extra names that may not have a value, using `getValue` will give an error when looking up the configuration. Before, only the property names that contained values were available (the ones defined by the user or defaults). To avoid that error, we now have to use `getConfigValue,` which, on the other hand, does not give an error when you have an expression that cannot be expanded, so in such cases, we force the `getValue` to give that error.